### PR TITLE
Friend fragment

### PR DESCRIPTION
**Pull request by andrew880**

### CODE REVIEW

**Review by ahominal — CHANGES_REQUESTED** [review]
There is still no test in this pull request ... Please make some at least for the UI

File: app/src/androidTest/java/com/sdp/swiftwallet/UiTest/MessageFragmentTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    @After
+    public void releaseIntents() {
+        Intents.release();
+    }
+
+    @Test
+    public void shows(){
+    }
```
Comment (0xdieman) [inline app/src/androidTest/java/com/sdp/swiftwallet/UiTest/MessageFragmentTest.java:60]: Isn't there a line missing ?

File: app/src/main/java/com/sdp/swiftwallet/presentation/friend/fragments/FriendItem.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+// Simple wallet model
+public class FriendItem {
+    public String address;
+
+    public FriendItem(String address) {
+        this.address = address;
+    }
+//    public void getName(double newBalance) {
+//        balance = newBalance;
+//    }
```
Comment (0xdieman) [inline app/src/main/java/com/sdp/swiftwallet/presentation/friend/fragments/FriendItem.java:12]: I think you could directly remove that

File: app/src/main/java/com/sdp/swiftwallet/presentation/main/fragments/MessageFragment.java [hunk 3]
```diff
@@ -1,25 +1,143 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+
+public class MessageFragment extends Fragment{
+    private FirebaseAuth mAuth;
```
Comment (0xdieman) [inline app/src/main/java/com/sdp/swiftwallet/presentation/main/fragments/MessageFragment.java:38]: You should be able to use hilt injection for this instance, at least for the next weeks

File: app/src/main/java/com/sdp/swiftwallet/presentation/main/fragments/MessageFragment.java [hunk 4]
```diff
@@ -1,25 +1,143 @@
...
+    }
+    private void setVisible(View view, boolean visible) {
+        view.findViewById(R.id.friendCode).setVisibility(visible ? View.VISIBLE : View.GONE);
+        view.findViewById(R.id.addFriendBtn).setVisibility(visible ? View.VISIBLE : View.GONE);
+        view.findViewById(R.id.addFriendStatus).setVisibility(View.GONE);
+    }
+
+    private void sendFriendRequest(View view, String userId, String friendId, String status) {
+        if (friendId.equals(userId)) {
+            ((TextView)view.findViewById(R.id.addFriendStatus)).setText("Cannot Add Yourself");
```
Comment (0xdieman) [inline app/src/main/java/com/sdp/swiftwallet/presentation/main/fragments/MessageFragment.java:90]: Those hardcoded strings would fit better in the values/strings file

**Comment by ahominal** [comment]
any update Andrew?

File: app/src/main/java/com/sdp/swiftwallet/presentation/main/fragments/MessageFragment.java (hunk 4 above)
Comment (0xdieman) [reply app/src/main/java/com/sdp/swiftwallet/presentation/main/fragments/MessageFragment.java:90]: I think

**Comment by Zotchh** [comment]
Closing the PR on this branch for now because it is outdated

**Pull request closed without merging**